### DISCUSSION TITLE
Handling setLocation on Android

### DISCUSSION THREAD

**Discussion by pikzen** [mobile-dev-inc/Maestro#262] [General]
Hey! I see that setLocation is available on iOS only. And of course, I would want to also make this work on Android, but there is no way to do this through adb, and I'd like to talk about the options you may have explored, and the ones you haven't.

One thing I can think of is having a Maestro APK to install on the device, with a Location Provider and the ACCESS_MOCK_LOCATION permission. On devices with developer tools, this would mean that the user could set the mock location provider to this MaestroMockLocator, and this service could (probably) receive intents to set the current location. Obviously impractical, but I struggle to see a better option.

**Comment by dmitry-zaitsev** [comment]
It is work-in-progress for Android - coming very soon!

**Comment by axelniklasson** [comment]
`setLocation` is now supported 🎉 Docs can be found [here](https://maestro.mobile.dev/api-reference/commands/setlocation). Thanks!
